R/sql_load.R

```diff
@@ -1,11 +1,24 @@
 get_sql_load <- function(path) {
-  statement <- paste(readLines(path), collapse = "\n")
+  ## It is not very nice that we have to read & parse this file to send it to
+  ## DB, ideally we could just execute the file directly on DB but DBI only
+  ## allows sending individual queries atm see
+  ## https://github.com/r-dbi/RPostgres/issues/213
+  sql <- readLines(path)
+  sql <- sql[!is_empty(sql)]
+  statement <- paste(sql, collapse = "\n")
+  statements <- unlist(strsplit(statement, ";", fixed = TRUE))
   function(con) {
     withCallingHandlers(
-      DBI::dbExecute(con, statement),
+      lapply(statements, function(cmd) {
+        DBI::dbExecute(con, cmd)
+      }),
       error = function(e) {
         e$message <- paste0("SQL error:\n", e$message)
         stop(e)
       })
   }
 }
+
+is_empty <- function(x) {
+  is.null(x) | is.na(x) | length(x) == 0 | trimws(x) == ""
```

Comment:
but looking at how this is used on l7 I don't see why it's needed or why `sql[nzchar(sql)]` is insufficient

Reply:
I want to vectorise it because `sql` is a vector on line 6 & 7. But `nzchar` is better - I just always forget that function exists.

Reply:
`is.null(x)` is scalar though so that would have been `||` otherwise it won't short circuit and you'll get odd answers from the following conditions:

```
x <- NULL
is.null(x) | is.na(x) | length(x) == 0 | trimws(x) == ""
# logical(0)
```